static/app/components/superuserAccessForm.tsx

```diff
@@ -85,14 +102,19 @@ class SuperuserAccessForm extends Component<Props, State> {
 
   handleU2fTap = async (data: Parameters<OnTapProps>[0]) => {
     const {api} = this.props;
-    try {
+
+    let auth_url = '/staff-auth/';
+    if (!this.props.hasStaff) {
       data.isSuperuserModal = true;
       data.superuserAccessCategory = this.state.superuserAccessCategory;
       data.superuserReason = this.state.superuserReason;
-      await api.requestPromise('/auth/', {method: 'PUT', data});
+      auth_url = '/auth/';
+    }
+    try {
+      await api.requestPromise(auth_url, {method: 'PUT', data});
       this.handleSuccess();
     } catch (err) {
-      this.setState({showAccessForms: true});
+      this.handleError(err);
```

Comment:
`this.handleError` also sets `showAccessForms` to `False`

Reply:
i think you mean setting to `true`. i'm also not sure why we weren't previously handling the error 🤷‍♀️

Reply:
Oh yes it's `true`